src/js/server/standaloneRenderer.js

```diff
@@ -11,159 +11,56 @@ var fluid = require("infusion");
 var gpii  = fluid.registerNamespace("gpii");
 fluid.registerNamespace("gpii.handlebars.standaloneRenderer");
 
-var handlebars = require("handlebars");
+require("../common/renderer");
 
 var fs   = require("fs");
 var path = require("path");
 
-fluid.require("%gpii-express"); // required to pick up `resolvePaths` function used below.
-
-gpii.handlebars.standaloneRenderer.addHelper = function (that, component) {
-    var key = component.options.helperName;
-    if (component.getHelper) {
-        that.helpers[key] = component.getHelper();
-    }
-    else {
-        fluid.fail("Can't register helper '" + key + "' because it doesn't have a getHelper() invoker.");
-    }
-};
-
-gpii.handlebars.standaloneRenderer.init = function (that) {
-    // Get rid of any existing compiled templates.
-    that.compiledTemplates = [];
-
-    // As with gradeNames, the right-most template directory that contains a given partial "wins".
-    fluid.each(that.templateDirs.reverse(), function (templateDir) {
-        // Register all partials found in the "partials" subdirectory relative to `options.templateDirs`;
-        var partialDir = path.resolve(templateDir, "partials");
-        fluid.each(fs.readdirSync(partialDir), function (filename) {
-            var partialPath = path.resolve(partialDir, filename);
-            var partialContent = fs.readFileSync(partialPath, "utf8");
-            var templateKey = filename.replace(/\.(handlebars|hbs)$/i, "");
-            handlebars.registerPartial(templateKey, partialContent);
+/**
+ *
+ * Examine one or more template directory paths found at `that.options.templateDirs` and load any template content found
+ * there.
+ *
+ * @param {Object} that - The renderer component itself.
+ *
+ */
+gpii.handlebars.standaloneRenderer.loadTemplateDirs = function (that) {
+    var handlebarsRegexp = /(.+)\.(hbs|handlebars)$/i;
+    var templateMap = {};
+    fluid.each(fluid.makeArray(that.options.templateDirs).reverse(), function (templateDirPath) {
+        var resolvedTemplateDirPath = fluid.module.resolvePath(templateDirPath);
+        fluid.each(["layouts", "pages", "partials"], function (subDir) {
+            var subDirPath = path.resolve(resolvedTemplateDirPath, subDir);
+            if (fs.existsSync(subDirPath)) {
+                var files = fs.readdirSync(subDirPath).filter(function (path) { return path.match(handlebarsRegexp); });
+                fluid.each(files, function (singleFile) {
+                    var matches = singleFile.match(handlebarsRegexp);
+                    var templateKey = matches[1];
+                    if (!fluid.get(templateMap, [subDir, templateKey])) {
+                        var filePath = path.resolve(subDirPath, singleFile);
+                        var templateContent = fs.readFileSync(filePath, "utf8");
+                        fluid.set(templateMap, [subDir, templateKey], templateContent);
+                    }
+                });
+            }
```

Comment:
The "apply" statement at the bottom here could also usefully be converted into a transactional "DELETE  plus ADD", this will be normalised in some future version of the ChangeApplier but right now you could usefully make a tiny utility to automate this

Reply:
Done, throughout.